### PR TITLE
:sparkles: Able to run controller inside the container platform with write access to root restrictions

### LINKED ISSUE #236: Support for OpenShift as Management Cluster
**User Story**

I am using OpenShift as my management cluster and encountered an issue when attempting to install the Helm chart on the managed cluster. The current version fails with an error related to writing to /.config and /etc/xdg folders. This is because, by default, pods on OpenShift run as non-privileged containers, which restricts write access to these directories.

**Current Workaround**
While changing the deployment to run as a privileged container is a potential workaround, it introduces security concerns and goes against best practices for running containers.

**Request**
Please modify the deployment configuration or provide an option to enable the Helm chart to function without requiring escalated privileges. This would involve ensuring that the necessary directories have appropriate permissions or altering the paths used by the application to directories writable by non-privileged containers.

**Additional Context**
The issue specifically arises because of the default security context in OpenShift, which does not permit write access to /.config and /etc/xdg. Adjusting the deployment to be compatible with OpenShift’s security policies will make the Helm chart more versatile and secure.

### PR DESCRIPTION
**Pull request by dmvolod**
**What this PR does / why we need it**:
This fix allows you to run a container with an operator on platforms that do not allow writing data to the root directory, such as OpenShift. 

The approach used is very simple, we mount /xdg as an emptyDir volume and redirect XDG_* variables used by helm there. This should work for upstream Kubernetes as well, but e2e validation is required.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #236

### CODE REVIEW

File: config/manager/manager.yaml
Comment (dmvolod) [inline config/manager/manager.yaml:79]: Probably xdg name and /xdg mount point will be better for it

Comment (dmvolod) [reply config/manager/manager.yaml:79]: Changed

File: config/manager/manager.yaml
Comment (Jont828) [inline config/manager/manager.yaml:83]:
```suggestion
        - emptyDir: {}
```

**Review by Jont828 — COMMENTED** [review]
Thanks for picking this up! Is the idea here just to mount the /xdg path and expect the user or management cluster to populate it? And are you able to test this with OpenShift and verify that it works?

**Comment by dmvolod** [comment]
@Jont828 could you please to point me to the deployed addon prodived helm container pod logs in failed CI.

**Comment by dmvolod** [comment]
> @Jont828 could you please to point me to the deployed addon prodived helm container pod logs in failed CI.

Oh, ran it locally and know why it happens. Will fix soon.

**Comment by Jont828** [comment]
@dmvolod Here's where the [pod logs](https://[storage.googleapis.com/kubernetes-ci-logs/pr-logs/pull/kubernetes-sigs_cluster-api-addon-provider-helm/322/pull-cluster-api-addon-provider-helm-e2e/1866905675351724032/artifacts/clusters/bootstrap/logs/caaph-system/caaph-controller-manager/caaph-controller-manager-b7fb6cfcb-wccwx/manager.log](https://storage.googleapis.com/kubernetes-ci-logs/pr-logs/pull/kubernetes-sigs_cluster-api-addon-provider-helm/322/pull-cluster-api-addon-provider-helm-e2e/1866905675351724032/artifacts/clusters/bootstrap/logs/caaph-system/caaph-controller-manager/caaph-controller-manager-b7fb6cfcb-wccwx/manager.log)) would be for the most recent run. When you're on the prow site, click on Artifacts and then it's in `/artifacts/clusters/bootstrap/logs/caaph-system/caaph-controller-manager/caaph-controller-manager-b7fb6cfcb-wccwx/manager.log`

File: config/manager/manager.yaml
Comment (Jont828) [reply config/manager/manager.yaml:89]: Sure, I think that seems reasonable as long it wouldn't collide with any files used by anything else in the container. Would we drop the `xdg` mount then?

Comment (dmvolod) [reply config/manager/manager.yaml]: I think yes, and /tmp will be enought for us as in my original proposal. Multiple paths /etc/xdg, $HOME and /tmp are utilizing for helm client. Will fix it soon.

**Review by Jont828 — APPROVED** [review]
/lgtm
/approve

Congrats on your first contribution!

**Comment by Jont828** [comment]
/approve cancel

Hold on, going to wait for CI to run before merging, but I don't think anything should break.

**Pull request merged**
